netutils/dhcpc/dhcpc.c

```diff
@@ -804,67 +811,70 @@ int dhcpc_request(FAR void *handle, FAR struct dhcpc_state *presult)
 
       /* Get the ACK/NAK response to the REQUEST (or timeout) */
 
-      result = recv(pdhcpc->sockfd, &pdhcpc->packet,
-                    sizeof(struct dhcp_msg), 0);
-      if (result >= 0)
+      recv_begin = clock();
+      do
         {
-          /* Parse the response */
+          result = recv(pdhcpc->sockfd, &pdhcpc->packet,
+                        sizeof(struct dhcp_msg), 0);
+          if (result >= 0)
+            {
+              /* Parse the response */
 
-          msgtype = dhcpc_parsemsg(pdhcpc, result, presult);
+              msgtype = dhcpc_parsemsg(pdhcpc, result, presult);
 
-          /* The ACK response means that the server has accepted
-           * our request and we have the lease.
-           */
+              /* The ACK response means that the server has accepted
+               * our request and we have the lease.
+               */
 
-          if (msgtype == DHCPACK)
-            {
-              ninfo("Received ACK\n");
-              state = STATE_HAVE_LEASE;
-            }
+              if (msgtype == DHCPACK)
+                {
+                  ninfo("Received ACK\n");
+                  state = STATE_HAVE_LEASE;
+                }
 
-          /* NAK means the server has refused our request.  Break out of
-           * this loop with state == STATE_HAVE_OFFER and send DISCOVER
-           * again
-           */
+              /* NAK means the server has refused our request */
 
-          else if (msgtype == DHCPNAK)
-            {
-              ninfo("Received NAK\n");
-              break;
-            }
+              else if (msgtype == DHCPNAK)
+                {
+                  ninfo("Received NAK\n");
+                  return ERROR;
```

Comment:
Attention here, it was "break" before this patch.
But since it no longer goes back to send DISCOVER, I changed it to "return ERROR"

Reply:
The "goes back to send DISCOVER" was invalid since https://github.com/apache/incubator-nuttx-apps/pull/862